src/mail.sh

```diff
@@ -115,8 +116,14 @@ function mail_send()
     cmd+=" --cc=\"$cc_recipients\""
   fi
 
-  pre_generate_patches "$commit_range" "$version"
-  patch_count="$(count_patches "$patch_cache")"
+  if [[ -z "$patch_source" ]]; then
+    pre_generate_patches "$commit_range" "$version"
+    patch_count="$(count_patches "$patch_cache")"
```

Comment:
Did you validate if your change does not break the original behavior? I mean, the one using the commit SHA.